plugins/modules/azure_rm_openshiftmanagedclusterkubeconfig_info.py

```diff
@@ -0,0 +1,227 @@
+                type='str', required=True
+            ),
+            name=dict(
+                type='str', required=True
+            ),
+            path=dict(
+                type='str', required=False
+            )
+        )
+
+        self.resource_group = None
+        self.name = None
+        self.path = None
+
+        self.results = dict(changed=False)
+        self.mgmt_client = None
+        self.state = None
+        self.url = None
+        self.status_code = [200]
+
+        self.query_parameters = {}
+        self.query_parameters['api-version'] = '2021-09-01-preview'
+        self.header_parameters = {}
+        self.header_parameters['Content-Type'] = 'application/json; charset=utf-8'
+
+        self.mgmt_client = None
+        super(AzureRMOpenShiftManagedClustersKubeconfigInfo, self).__init__(self.module_arg_spec, supports_check_mode=True, supports_tags=False)
+
+    def exec_module(self, **kwargs):
+
+        for key in self.module_arg_spec:
+            setattr(self, key, kwargs[key])
+
+        self.mgmt_client = self.get_mgmt_svc_client(GenericRestClient, is_track2=True,
+                                                    base_url=self._cloud_environment.endpoints.resource_manager)
+        self.results = self.get_kubeconfig()
+        if self.path and self.path_is_valid():
+            self.write_kubeconfig_to_file()
+        return self.results
+
+    def get_kubeconfig(self):
+        response = None
+        results = {}
+        # prepare url
+        self.url = ('/subscriptions' +
+                    '/{{ subscription_id }}' +
+                    '/resourceGroups' +
+                    '/{{ resource_group }}' +
+                    '/providers' +
+                    '/Microsoft.RedHatOpenShift' +
+                    '/openShiftClusters' +
+                    '/{{ cluster_name }}' +
+                    '/listAdminCredentials')
+        self.url = self.url.replace('{{ subscription_id }}', self.subscription_id)
+        self.url = self.url.replace('{{ resource_group }}', self.resource_group)
+        self.url = self.url.replace('{{ cluster_name }}', self.name)
+        self.log("Fetch for kubeconfig from the cluster.")
+        try:
+            response = self.mgmt_client.query(self.url,
+                                              'POST',
+                                              self.query_parameters,
+                                              self.header_parameters,
+                                              None,
+                                              self.status_code,
+                                              600,
+                                              30)
+            results = json.loads(response.body())
+        except Exception as e:
+            self.log('Could not get info for @(Model.ModuleOperationNameUpper).')
+        return self.format_item(results)
+
+    def format_item(self, item):
+        d = {
+            'kubeconfig': item['kubeconfig'],
```

Comment:
An empty dictionary is returned when the resource you fetched is non-existent, so it is recommended to change it as follows!
```suggestion
            'kubeconfig': item.get('kubeconfig'),
```

Reply:
Done